integration/integration_test.go

```diff
@@ -1423,7 +1424,7 @@ func testIPPropagation(t *testing.T, suite *integrationTestSuite) {
 					return nil, trace.Wrap(err)
 				}
 
-				local.set(conn.LocalAddr())
+				local.add(conn.LocalAddr())
```

Comment:
I don't follow what the changes here have to do with the rest of the PR.

(It looks OK by itself, I just missing how it fits.)